### ISSUE TITLE
Godot 3.5.rc1 for Mono+Win64 crashes shortly after startup on Windows 10

### ISSUE THREAD

**Issue by David1Socha** [godotengine/godot#61176]
### Godot version

3.5.rc1

### System information

Windows 10

### Issue description

I downloaded the Godot 3.5 rc1 x64 windows build, and was trying to see if my project from 3.4.4 could open successfully. Whenever I ran Godot_v3.5-rc1_mono_win64.exe, I saw the startup splash, and the project selection screen, but the screen was unresponsive, and exited with an error about a second later. 

Confirmed that this issue does not occur on 3.4.4, nor on 3.5.beta5 . Seems to have been introduced with 3.5-rc1

### Steps to reproduce

Download Godot 3.5-rc1 Win64 mono build (I used the link from the announcement page at https://downloads.tuxfamily.org/godotengine/3.5/rc1/Godot_v3.5-rc1_win64.exe.zip )
Unzip build.
Run Godot_v3.5-rc1_mono_win64.exe.
At this point, Godot crashes shortly after startup.

### Minimal reproduction project

Seems to be failing across the board during startup, at least for my machine. Doesn't seem to be project-specific since this error happens before any project is loaded.

**Comment by akien-mga** [comment]
Probably related to changes in the official buildsystem, I upgraded from:

Fedora 34, MinGW-GCC 10.2.1, Mono 6.12.0.158

to:

Fedora 35, MinGW-GCC 11.2.1, Mono 6.12.0.178

**Comment by akien-mga** [comment]
The Mono logs indicate that it's searching for and can't find `BCrypt.dll`.

That's because of https://github.com/godotengine/godot-mono-builds/pull/47, which used to be reverted in my previous build containers due to it breaking macOS ARM64, but apparently it's also not working for Windows either... So https://github.com/godotengine/godot-mono-builds/pull/65 was not a good option and we need a proper revert of both PRs now.

**Comment by MadFlyFish** [comment]
This is quite a serious problem.
Need more testing bofore release.

**Comment by akien-mga** [comment]
@MadFlyFish A Release Candidate *is* the testing before release.

**Comment by akien-mga** [comment]
> The Mono logs indicate that it's searching for and can't find `BCrypt.dll`.
> 
> That's because of [godotengine/godot-mono-builds#47](https://github.com/godotengine/godot-mono-builds/pull/47), which used to be reverted in my previous build containers due to it breaking macOS ARM64, but apparently it's also not working for Windows either... So [godotengine/godot-mono-builds#65](https://github.com/godotengine/godot-mono-builds/pull/65) was not a good option and we need a proper revert of both PRs now. @neikeq

Seems I was maybe a bit quick to jump to conclusions, I tried a new build after those reverts and it's still crashing. Then I noticed that the 3.5 beta 5 Mono logs (version which works) also spam a lot about attempts to find `BCrypt`, that was a red herring.

Here's the difference between the start logs for 3.5 beta 5 and 3.5 RC 1 (binaries renamed to have the same path each time to reduce the diff):
```diff
$ diff -aur 2022-05-19_16.54.06_4140.log 2022-05-19_16.54.22_4120.log
--- 2022-05-19_16.54.06_4140.log        2022-05-19 16:54:10.016498200 +0200
+++ 2022-05-19_16.54.22_4120.log        2022-05-19 16:54:23.937673900 +0200
@@ -1,61 +1,61 @@
 Config attempting to parse: 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/etc\mono\config'. (in domain Mono, info)
-Image addref mscorlib[000000001b289ac0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/mscorlib.dll[000000001b3b0740]: 2 (in domain Mono, info)
+Image addref mscorlib[000001b2e29eee00] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/mscorlib.dll[000001b2e2b0aaf0]: 2 (in domain Mono, info)
 Prepared to set up assembly 'mscorlib' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/mscorlib.dll) (in domain Mono, info)
 AOT: image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/mscorlib.dll.dll' not found: The system cannot find the file specified.
  (in domain Mono, info)
 AOT: image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/aot-cache/amd64/mscorlib.dll.dll' not found: The system cannot find the path specified.
  (in domain Mono, info)
-Assembly mscorlib[000000001b289ac0] added to domain GodotEngine.RootDomain, ref_count=1 (in domain Mono, info)
-Assembly mscorlib[000000001b289ac0] added to domain GodotEngine.Domain.Scripts, ref_count=2 (in domain Mono, info)
-Image addref GodotSharp[000000001b2845d0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll[000000001b42fdd0]: 2 (in domain Mono, info)
+Assembly mscorlib[000001b2e29eee00] added to domain GodotEngine.RootDomain, ref_count=1 (in domain Mono, info)
+Assembly mscorlib[000001b2e29eee00] added to domain GodotEngine.Domain.Scripts, ref_count=2 (in domain Mono, info)
+Image addref GodotSharp[000001b2e29e9be0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll[000001b2e2bab2c0]: 2 (in domain Mono, info)
 Prepared to set up assembly 'GodotSharp' (C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll) (in domain Mono, info)
-Assembly GodotSharp[000000001b2845d0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly GodotSharp[000001b2e29e9be0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
 Requesting loading reference 0 (of 1) of C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll asmctx DEFAULT, looking for netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (in domain Mono, info)
 Config attempting to parse: ''. (in domain Mono, info)
-Image addref netstandard[000000001b284780] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll[00000000262460b0]: 2 (in domain Mono, info)
+Image addref netstandard[000001b2e29e9910] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll[000001b2ebb5a050]: 2 (in domain Mono, info)
 Prepared to set up assembly 'netstandard' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll) (in domain Mono, info)
-Assembly netstandard[000000001b284780] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Assembly Ref addref GodotSharp[000000001b2845d0] -> netstandard[000000001b284780]: 2 (in domain Mono, info)
+Assembly netstandard[000001b2e29e9910] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly Ref addref GodotSharp[000001b2e29e9be0] -> netstandard[000001b2e29e9910]: 2 (in domain Mono, info)
 Requesting loading reference 0 (of 16) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll asmctx DEFAULT, looking for mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref netstandard[000000001b284780] -> mscorlib[000000001b289ac0]: 3 (in domain Mono, info)
-Image addref GodotSharpEditor[000000001b284810] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll[000000002627b120]: 2 (in domain Mono, info)
+Assembly Ref addref netstandard[000001b2e29e9910] -> mscorlib[000001b2e29eee00]: 3 (in domain Mono, info)
+Image addref GodotSharpEditor[000001b2e29e8290] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll[000001b2ebba0950]: 2 (in domain Mono, info)
 Prepared to set up assembly 'GodotSharpEditor' (C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll) (in domain Mono, info)
-Assembly GodotSharpEditor[000000001b284810] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly GodotSharpEditor[000001b2e29e8290] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
 Requesting loading reference 0 (of 2) of C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll asmctx DEFAULT, looking for netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 (in domain Mono, info)
-Assembly Ref addref GodotSharpEditor[000000001b284810] -> netstandard[000000001b284780]: 3 (in domain Mono, info)
+Assembly Ref addref GodotSharpEditor[000001b2e29e8290] -> netstandard[000001b2e29e9910]: 3 (in domain Mono, info)
 Requesting loading reference 2 (of 16) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll (in domain Mono, info)
 Loading reference 2 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll asmctx DEFAULT, looking for System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Image addref System[000000001b283cd0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll[00000000262bf010]: 2 (in domain Mono, info)
+Image addref System[000001b2e29e8e60] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll[000001b2ebbc96e0]: 2 (in domain Mono, info)
 Prepared to set up assembly 'System' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll) (in domain Mono, info)
-Assembly System[000000001b283cd0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Assembly Ref addref netstandard[000000001b284780] -> System[000000001b283cd0]: 2 (in domain Mono, info)
+Assembly System[000001b2e29e8e60] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly Ref addref netstandard[000001b2e29e9910] -> System[000001b2e29e8e60]: 2 (in domain Mono, info)
 Requesting loading reference 0 (of 6) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll asmctx DEFAULT, looking for mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System[000000001b283cd0] -> mscorlib[000000001b289ac0]: 4 (in domain Mono, info)
+Assembly Ref addref System[000001b2e29e8e60] -> mscorlib[000001b2e29eee00]: 4 (in domain Mono, info)
 Requesting loading reference 2 (of 6) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll (in domain Mono, info)
 Loading reference 2 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll asmctx DEFAULT, looking for System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a (in domain Mono, info)
-Image addref System.Configuration[000000001b284030] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll[0000000026334d10]: 2 (in domain Mono, info)
+Image addref System.Configuration[000001b2e29e84d0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll[000001b2ebc49820]: 2 (in domain Mono, info)
 Prepared to set up assembly 'System.Configuration' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll) (in domain Mono, info)
-Assembly System.Configuration[000000001b284030] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Assembly Ref addref System[000000001b283cd0] -> System.Configuration[000000001b284030]: 2 (in domain Mono, info)
+Assembly System.Configuration[000001b2e29e84d0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly Ref addref System[000001b2e29e8e60] -> System.Configuration[000001b2e29e84d0]: 2 (in domain Mono, info)
 Requesting loading reference 0 (of 4) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll asmctx DEFAULT, looking for mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System.Configuration[000000001b284030] -> mscorlib[000000001b289ac0]: 5 (in domain Mono, info)
+Assembly Ref addref System.Configuration[000001b2e29e84d0] -> mscorlib[000001b2e29eee00]: 5 (in domain Mono, info)
 Requesting loading reference 2 (of 4) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll (in domain Mono, info)
 Loading reference 2 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll asmctx DEFAULT, looking for System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System.Configuration[000000001b284030] -> System[000000001b283cd0]: 3 (in domain Mono, info)
+Assembly Ref addref System.Configuration[000001b2e29e84d0] -> System[000001b2e29e8e60]: 3 (in domain Mono, info)
 Requesting loading reference 1 (of 4) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll (in domain Mono, info)
 Loading reference 1 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll asmctx DEFAULT, looking for System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Image addref System.Xml[000000001b2848a0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll[000000002625b800]: 2 (in domain Mono, info)
+Image addref System.Xml[000001b2e29e9c70] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll[000001b2ebb82030]: 2 (in domain Mono, info)
 Prepared to set up assembly 'System.Xml' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll) (in domain Mono, info)
-Assembly System.Xml[000000001b2848a0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Assembly Ref addref System.Configuration[000000001b284030] -> System.Xml[000000001b2848a0]: 2 (in domain Mono, info)
+Assembly System.Xml[000001b2e29e9c70] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Assembly Ref addref System.Configuration[000001b2e29e84d0] -> System.Xml[000001b2e29e9c70]: 2 (in domain Mono, info)
 Requesting loading reference 0 (of 3) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll (in domain Mono, info)
 Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll asmctx DEFAULT, looking for mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System.Xml[000000001b2848a0] -> mscorlib[000000001b289ac0]: 6 (in domain Mono, info)
+Assembly Ref addref System.Xml[000001b2e29e9c70] -> mscorlib[000001b2e29eee00]: 6 (in domain Mono, info)
 DllImport attempting to load: 'BCrypt.dll'. (in domain Mono, info)
 DllImport error loading library 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5\BCrypt.dll': 'The system cannot find the file specified.
 '. (in domain Mono, info)
@@ -84,60 +84,12 @@
 Found as 'GetFileAttributesExW'. (in domain Mono, info)
 Requesting loading reference 2 (of 3) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll (in domain Mono, info)
 Loading reference 2 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll asmctx DEFAULT, looking for System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System.Xml[000000001b2848a0] -> System[000000001b283cd0]: 4 (in domain Mono, info)
+Assembly Ref addref System.Xml[000001b2e29e9c70] -> System[000001b2e29e8e60]: 4 (in domain Mono, info)
 Requesting loading reference 3 (of 6) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll (in domain Mono, info)
 Loading reference 3 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll asmctx DEFAULT, looking for System.Xml, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System[000000001b283cd0] -> System.Xml[000000001b2848a0]: 3 (in domain Mono, info)
-Image addref GodotTools[000000001b282920] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.dll[000000002622e4a0]: 2 (in domain Mono, info)
+Assembly Ref addref System[000001b2e29e8e60] -> System.Xml[000001b2e29e9c70]: 3 (in domain Mono, info)
+Image addref GodotTools[000001b2e29e9880] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.dll[000001b2ebb427b0]: 2 (in domain Mono, info)
 Prepared to set up assembly 'GodotTools' (C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.dll) (in domain Mono, info)
-Assembly GodotTools[000000001b282920] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Image addref GodotTools.ProjectEditor[000000001b282ad0] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.ProjectEditor.dll[0000000026259010]: 2 (in domain Mono, info)
+Assembly GodotTools[000001b2e29e9880] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
+Image addref GodotTools.ProjectEditor[000001b2e29ea180] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.ProjectEditor.dll[000001b2ebc0b010]: 2 (in domain Mono, info)
 Prepared to set up assembly 'GodotTools.ProjectEditor' (C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.ProjectEditor.dll) (in domain Mono, info)
-Assembly GodotTools.ProjectEditor[000000001b282ad0] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Requesting loading reference 1 (of 16) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll (in domain Mono, info)
-Loading reference 1 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll asmctx DEFAULT, looking for System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Image addref System.Core[00000000270bf720] (asmctx DEFAULT) -> C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Core.dll[000000002f475a30]: 2 (in domain Mono, info)
-Prepared to set up assembly 'System.Core' (C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Core.dll) (in domain Mono, info)
-Assembly System.Core[00000000270bf720] added to domain GodotEngine.Domain.Scripts, ref_count=1 (in domain Mono, info)
-Assembly Ref addref netstandard[000000001b284780] -> System.Core[00000000270bf720]: 2 (in domain Mono, info)
-Requesting loading reference 0 (of 2) of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Core.dll (in domain Mono, info)
-Loading reference 0 of C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Core.dll asmctx DEFAULT, looking for mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 (in domain Mono, info)
-Assembly Ref addref System.Core[00000000270bf720] -> mscorlib[000000001b289ac0]: 7 (in domain Mono, info)
-GC_MAJOR: (user request) time 1.21ms, stw 1.25ms los size: 1024K in use: 64K (in domain Mono, info)
-GC_MAJOR_SWEEP: major size: 1040K in use: 186K (in domain Mono, info)
-GC_MAJOR: (user request) time 0.37ms, stw 0.41ms los size: 1024K in use: 64K (in domain Mono, info)
-GC_MAJOR_SWEEP: major size: 1056K in use: 187K (in domain Mono, info)
-GC_MAJOR: (user request) time 0.36ms, stw 0.43ms los size: 1024K in use: 64K (in domain Mono, info)
-GC_MAJOR_SWEEP: major size: 1056K in use: 187K (in domain Mono, info)
-GC_MINOR: (user request) time 0.15ms, stw 0.20ms promoted 0K major size: 1056K in use: 187K los size: 1024K in use: 64K (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly mscorlib[000000001b289ac0], ref_count=7 (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly GodotSharp[000000001b2845d0], ref_count=1 (in domain Mono, info)
-Unloading assembly GodotSharp [000000001b2845d0]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharp.dll [000000001b42fdd0]. (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly netstandard[000000001b284780], ref_count=2 (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly GodotSharpEditor[000000001b284810], ref_count=1 (in domain Mono, info)
-Unloading assembly GodotSharpEditor [000000001b284810]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Api/Debug/GodotSharpEditor.dll [000000002627b120]. (in domain Mono, info)
-Unloading assembly netstandard [000000001b284780]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/Facades/netstandard.dll [00000000262460b0]. (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly System[000000001b283cd0], ref_count=3 (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly System.Configuration[000000001b284030], ref_count=2 (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly System.Xml[000000001b2848a0], ref_count=3 (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly GodotTools[000000001b282920], ref_count=1 (in domain Mono, info)
-Unloading assembly GodotTools [000000001b282920]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.dll [000000002622e4a0]. (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly GodotTools.ProjectEditor[000000001b282ad0], ref_count=1 (in domain Mono, info)
-Unloading assembly GodotTools.ProjectEditor [000000001b282ad0]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Tools/GodotTools.ProjectEditor.dll [0000000026259010]. (in domain Mono, info)
-Unloading domain GodotEngine.Domain.Scripts[000000001b3f8950], assembly System.Core[00000000270bf720], ref_count=1 (in domain Mono, info)
-Unloading assembly System.Core [00000000270bf720]. (in domain Mono, info)
-Unloading image C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Core.dll [000000002f475a30]. (in domain Mono, info)
-GC_MAJOR: (user request) time 0.19ms, stw 0.28ms los size: 0K in use: 0K (in domain Mono, info)
-GC_MAJOR_SWEEP: major size: 240K in use: 2K (in domain Mono, info)
-GC_MAJOR: (user request) time 0.20ms, stw 0.24ms los size: 0K in use: 0K (in domain Mono, info)
-GC_MAJOR_SWEEP: major size: 240K in use: 2K (in domain Mono, info)
-Unloading domain GodotEngine.RootDomain[000000001b3aa8a0], assembly mscorlib[000000001b289ac0], ref_count=4 (in domain Mono, info)
-Assembly image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.dll' [00000000262bf010] still loaded at shutdown. (in domain Mono, info)
-Assembly image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/mscorlib.dll' [000000001b3b0740] still loaded at shutdown. (in domain Mono, info)
-Assembly image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Xml.dll' [000000002625b800] still loaded at shutdown. (in domain Mono, info)
-Assembly image 'C:/Users/Remi/Downloads/Godot/GodotSharp/Mono/lib/mono/4.5/System.Configuration.dll' [0000000026334d10] still loaded at shutdown. (in domain Mono, info)
```

So it just stops abruptly at the time where it's supposed to load the ProjectEditor assembly. Not sure what to make of it.

**Comment by akien-mga** [comment]
> _Edit:_ Well the difference between the two is that my "fixed" 3.5 RC 1 build does lack `libmono-btls-shared.dll` which 3.5 beta 5 had. I'm confused... I have to double check that I built the right version for my test.

Seems like my build containers differ indeed, here's the `3.x-mono-6.12.0.158` based on Fedora 34 and Mono 6.12.0.158 which we used up to 3.5 beta 5 included:
```
# find -name "libmono-btls*"
./usr/lib/libmono-btls-shared.so
./root/mono-configs/desktop-windows-x86-release/mono/btls/build-shared/libmono-btls-shared.dll
./root/mono-configs/desktop-windows-x86-release/mono/btls/build-shared/libmono-btls-shared.dll.a
./root/mono-configs/desktop-windows-x86_64-release/mono/btls/build-shared/libmono-btls-shared.dll
./root/mono-configs/desktop-windows-x86_64-release/mono/btls/build-shared/libmono-btls-shared.dll.a
./root/mono-installs/desktop-windows-x86-release/bin/libmono-btls-shared.dll
./root/mono-installs/desktop-windows-x86-release/lib/libmono-btls-shared.dll.a
./root/mono-installs/desktop-windows-x86_64-release/bin/libmono-btls-shared.dll
./root/mono-installs/desktop-windows-x86_64-release/lib/libmono-btls-shared.dll.a
```

While in the new Fedora 35 based one with Mono 6.12.0.179, we seem to lack btls in the godot-mono-builds output:
```
# find -name "libmono-btls*"
./usr/lib/libmono-btls-shared.so
```

Yet I don't see any difference in the `godot-mono-builds` code used for each of them (they're based on different commits but after my reverts they should be equivalent...).

**Comment by akien-mga** [comment]
Alright, the btls stuff was a red herring. I found that my old container did have `libmono-btls-shared.dll` in the end, so that probably means that I built the Windows container before reverting the change in godot-mono-builds which I then committed to build the macOS ARM64 version... duh.

On the other hand it doesn't seem to make a difference for this issue. I have these three tests builds:

- Fedora 34, Mono 6.12.0.158, with btls-shared (works): https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc1_mono_win64-f34-mono158-btls.zip
- Fedora 35, Mono 6.12.0.178, with btls-shared (broken): https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc1_mono_win64-f35-mono178-btls.zip
- Fedora 35, Mono 6.12.0.179, without btls-shared (broken): https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc1_mono_win64-f35-mono179-nobtls.zip

So I'll have to do more tests and see if the issue comes from the difference in system libraries between Fedora 34 and Fedora 35, or from one of the 20 Mono commits between 6.12.0.158 and 6.12.0.178.

**Comment by rafaelgdp** [comment]
I am here to confirm this also happens to me exactly as described by @David1Socha . I'm on a **Windows 11 Notebook**. Godot Mono Beta 5 worked just fine on my machine. This crash was introduced in the RC1 mono build.

**Comment by akien-mga** [comment]
So I did a few more test builds to add to the above:

- Fedora 34, Mono 6.12.0.179, without btls-shared (works): https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc1_mono_win64-f34-mono179-nobtls.zip
- Fedora 36, Mono 6.12.0.179, without btls-shared (works): https://downloads.tuxfamily.org/godotengine/testing/Godot_v3.5-rc1_mono_win64-f36-mono179-nobtls.zip

So crossing those results it's pretty clear that the btls thing is unrelated to this crash, and that the upgrade from Mono 6.12.0.158 to 6.12.0.179 is not an issue either (works on Fedora 34 like the old version).

Since Fedora 34 *and* Fedora 36 both work fine, I'm now pretty sure that it's a crash caused by a bug in MinGW, GCC and/or binutils in Fedora 35, which is related to #51070. @MmAaXx500 debugged it well and found indeed that the versions in Fedora 35 are problematic - it doesn't seem to affect the `3.x` standard build for some reason, but apparently it's affecting Mono itself.

So I'll just use Fedora 36 for the next builds, it seems to work fine in my tests. (I initially updated everything to Fedora 36 but then went down to Fedora 35 due to issues with LLVM 14 for osxcross... I'll figure something out for macOS/iOS builds, worst case those can stay on Fedora 35.)

**Comment by akien-mga** [comment]
Fixed in 3.5 RC 2.
